File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int v : nums){
            map.put(v, map.getOrDefault(v, 0)+1);
        }
        for(int k: nums){
            if(map.containsKey(k)){
                int temp = map.get(k);
                if(temp == 1) return k;
            }
        }
        
        return -1;
    }
}

File: 0136-single-number/0136-single-number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        freq = {}
        for num in  nums:
            freq[num] = freq.get(num, 0)+1

        for key, value in freq.items():
            if value == 1:
                return key
        return -1
        